### ISSUE TITLE
Implement `$$expect` builtin

### ISSUE THREAD

**Issue by data-man** [c3lang/c3c#647]
LLVM provides some useful builtins for this:
```llvm
declare i1 @llvm.expect.i1(i1 <val>, i1 <expected_val>)
declare i32 @llvm.expect.i32(i32 <val>, i32 <expected_val>)
declare i64 @llvm.expect.i64(i64 <val>, i64 <expected_val>)

declare i1 @llvm.expect.with.probability.i1(i1 <val>, i1 <expected_val>, double <prob>)
declare i32 @llvm.expect.with.probability.i32(i32 <val>, i32 <expected_val>, double <prob>)
declare i64 @llvm.expect.with.probability.i64(i64 <val>, i64 <expected_val>, double <prob>)
```
Proposals:
* add `$$expect` builtin (using `@llvm.expect.with.probability.i1`)
* add macros to `std.core.builtin`:
```rust
macro bool @expect(bool x, float probability = 1.0) @builtin => $$expect(x, probability);
macro bool @likely(bool x) @builtin => $$expect(x, 1.0);
macro bool @unlikely(bool x) @builtin => $$expect(x, 0.0);
```

Usage:
```c++
if (@expect(a < 3, 0.3))
{...}
else if (@expect(a >= 3, 0.5))
     {...}

if (@likely(a > 0)) {...}

if (@unlikely(a < 0)) {...}
```

Probably `llvm.expect.*` intrinsics can be used internally to hint the LLVM optimizer.

**Comment by lerno** [comment]
The alternative is:
```c
if (a > 0 @likely) { ... }
```
Which seems nicer?

**Comment by data-man** [comment]
I prefer my proposal. :)
expect/likely/unlikely are widely used in C projects, so porting them to C3 will more comfortable.

**Comment by data-man** [comment]
Implemented in 6b928c7a3dae786f2053fbed7b50b7e282bc4648, thank you!

My vision in #727. :)
